bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,273 @@
+import org.openhab.binding.openthermgateway.internal.GatewayCommand;
+import org.openhab.binding.openthermgateway.internal.GatewayCommandCode;
+import org.openhab.binding.openthermgateway.internal.Message;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayCallback;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConfiguration;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConnector;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewaySocketConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenThermGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arjen Korevaar - Initial contribution
+ */
+@NonNullByDefault
+public class OpenThermGatewayHandler extends BaseThingHandler implements OpenThermGatewayCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenThermGatewayHandler.class);
+
+    private @Nullable OpenThermGatewayConfiguration config;
+
+    private @Nullable OpenThermGatewayConnector connector;
+
+    private boolean connecting = false;
+
+    private boolean explicitDisconnect = false;
+
+    public OpenThermGatewayHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenTherm Gateway handler for uid '{}'", getThing().getUID());
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Initializing");
+
+        config = getConfigAs(OpenThermGatewayConfiguration.class);
+
+        connect();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        if (!(command instanceof RefreshType)) {
+            String channel = channelUID.getId();
+            String code = getGatewayCodeFromChannel(channel);
+
+            GatewayCommand gatewayCommand = null;
+
+            if (command instanceof QuantityType<?>) {
+                QuantityType<?> quantityType = ((QuantityType<?>) command).toUnit(SIUnits.CELSIUS);
+
+                if (quantityType != null) {
+                    double value = quantityType.doubleValue();
+                    gatewayCommand = GatewayCommand.parse(code, Double.toString(value));
+                }
+            }
+
+            if (gatewayCommand == null) {
+                gatewayCommand = GatewayCommand.parse(code, command.toFullString());
+            }
+
+            if (checkConnection()) {
+                @Nullable
+                OpenThermGatewayConnector conn = connector;
+
+                if (conn != null) {
+                    conn.sendCommand(gatewayCommand);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void connecting() {
+        connecting = true;
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Connecting");
```

Comment:
Does connecting mean that is not connected? Or should it just be UKNOWN? The status which we talk about is the status of actual thing not of the connection. 

So if you connected you know its ONLINE, if you fail to connect you can assume its OFFLINE or that there is some configuration error. But if you haven't tried you can nog know whether the thing is OFFLINE.

Reply:
For your reference see also: https://www.openhab.org/docs/concepts/things.html#thing-status

Reply:
Changed